### PR TITLE
add min_depth to toc() helper

### LINKED ISSUE #3288: TOC helper: add min_depth
## Enhancement

On the TOC helper (https://hexo.io/docs/helpers#toc), it would be great to add a `min_depth` option so we are able to set a minimum heading depth of generated toc.

This way we will be able to exclude h1 from the TOC for example.

### PR DESCRIPTION
**Pull request by noraj**
## What does it do?

fix #3288

## How to test

```sh
git clone -b BRANCH https://github.com/noraj/hexo.git
cd hexo
git checkout toc_min_depth
npm install
npm test
npm run eslint
```

## Pull request tasks

- [ ] Add test cases for the changes. [WIP] tests for toc plugin seem complex, so I begin to write them but I wasn't able to finish.
- [ ] Passed the CI test. No, because tests are not finished.

### CODE REVIEW

File: lib/plugins/helper/toc.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ function tocHelper(str, options = {}) {
 
   const $ = cheerio.load(str);
   const headingsMaxDepth = Object.prototype.hasOwnProperty.call(options, 'max_depth') ? options.max_depth : 6;
-  const headingsSelector = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].slice(0, headingsMaxDepth).join(',');
+  const headingsMinDepth = Object.prototype.hasOwnProperty.call(options, 'min_depth') ? options.min_depth : 1;
```
Comment (SukkaW) [inline lib/plugins/helper/toc.js]: It is better to use `Object.assign` to handle multiple default config.

**Comment by curbengh** [comment]
Is this compatible with https://github.com/hexojs/hexo/pull/3850?

**Comment by SukkaW** [comment]
@curbengh I could bring up #3850 (with `hexo-util`'s `tocObj()`) over this PR then.

**Comment by curbengh** [comment]
My understanding is that [`tocObj()`](https://github.com/hexojs/hexo-util/pull/137) implemented min_depth, so this PR should be superseded by https://github.com/hexojs/hexo/pull/3850.

**Pull request closed without merging**